cmd/lakectl/cmd/fs_upload.go

```diff
@@ -32,13 +32,16 @@ var fsUploadCmd = &cobra.Command{
 
 		ctx, stop := signal.NotifyContext(ctx, os.Interrupt, os.Kill)
 		defer stop()
-
-		stat, err := os.Stat(source)
-		if err != nil {
-			Die("failed to stat source", 1)
+		ignoreRecessive := true // Ignore recursive if source is the standard input or a file (not a directory)
```

Comment:
This should probably be a function: it already needs it's own documentation, might as well give it a name and a godoc.